src/test/util/mod.rs

```diff
@@ -188,8 +194,8 @@ pub async fn drop_collection(coll: &Collection) {
 
 pub fn parse_version(version: &str) -> (u64, u64) {
```

Comment:
I think this method can be replaced to use `semver::Version`. We're using it already in `TestClient`.

Reply:
I think we need the more flexible parser since the tests specify strings with and without patch versions (i.e. both "4.0" and "4.1.7") while the `Version::parse` function only takes strings with patch versions.

Reply:
Potential hacky workaround: if there is only one '.', append ".0" to the string and then pass to semver.

Reply:
This function is being called to parse the versions so they can be passed into `server_version_gt` or `server_version_lt` which both only compare against the major/minor versions, so using `Version::parse` to parse out the patch version won't add any extra accuracy. Maybe we should introduce an `Option<u64>` patch parameter to the gt/lt methods?

Reply:
I think eventually the gt/lt methods can just be removed. `semver::Version` is `Ord`, so we can just use the regular comparison operators on the type itself. Also note that `TestClient.server_version` is public, so we can just do `parsed_version > client.server_version` etc.